### ISSUE TITLE
Importing DistributedSampler changes matplotlib backend

### ISSUE THREAD

**Issue by Lucas-Steinmann** [open-mmlab/mmpose#776]
**Checklist**

1. [x] I have searched related issues but cannot get the expected help.
1. [x] The bug has not been fixed in the latest version.

**Describe the bug**

After starting to use mmpose, suddenly my matplotlib figures stopped showing up. This error was produced:
```
scripts/test.py:9: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.
```

So I strategically placed this snippet:
```
plt.imshow(np.random.random((10, 10)), interpolation='nearest')
plt.show()
```
at various positions in my script to locate the error.
After some time I found that while this works:
```
import numpy as np
import matplotlib.pyplot as plt
plt.imshow(np.random.random((10, 10)), interpolation='nearest')
plt.show()
from mmpose.datasets.samplers import DistributedSampler
```
this does not:

```
import numpy as np
import matplotlib.pyplot as plt
from mmpose.datasets.samplers import DistributedSampler
plt.imshow(np.random.random((10, 10)), interpolation='nearest')
plt.show()
```

I cannot fathom what in `DistributedSampler` would change the matplotlib backend, since the class does not do a lot. But as this script indicates, it does:

```
import matplotlib
print(matplotlib.get_backend())
from mmpose.datasets.samplers import DistributedSampler
print(matplotlib.get_backend())
```
produces the follwing output:
```
TkAgg
agg
```
**Reproduction**

- What command or script did you run?

```
import matplotlib
print(matplotlib.get_backend())
from mmpose.datasets.samplers import DistributedSampler
print(matplotlib.get_backend())
```

- What config did you run? **n.a.**

- Did you make any modifications on the code or config? Did you understand what you have modified? I added a new dataset and configs, but did not change anything else. I don't think my changes, did create the bug, which is described here.

- What dataset did you use? **n.a.**

**Environment**

pytorch install via conda

```
sys.platform: linux
Python: 3.8.10 (default, Jun  4 2021, 15:09:15) [GCC 7.5.0]
CUDA available: True
GPU 0: GeForce GTX 1080 Ti
CUDA_HOME: /usr/local/cuda/
NVCC: Cuda compilation tools, release 10.1, V10.1.105
GCC: gcc (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
PyTorch: 1.8.1
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.2-Product Build 20210312 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.7.0 (Git Hash 7aed236906b1f7a05c0917e5257a1af05e9ff683)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.1, CUDNN_VERSION=7.6.3, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.8.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.9.1
OpenCV: 4.5.2
MMCV: 1.3.8
MMCV Compiler: GCC 5.4
MMCV CUDA Compiler: 10.1
MMPose: 0.16.0+1200dd0
```
**Error traceback**

If applicable, paste the error traceback here.
**n.a.**

**Bug fix**

I did not find a reason why this behavior could be created. I think I'm going mad, this doesn't make any sense. **Please, can someone reproduce the bug, before I further try to fix it.** I worked around the issue by resetting the backend after importing via:
```
matplotlib.use('TkAgg')
```
But this is rather a dirty fix than a solution.

**Comment by ly015** [comment]
The matplotlib backend was set by xtcocotools here:

**Comment by jin-s13** [comment]
This bug will be fixed in xtcocotools==1.9. Please install the latest xtcocotools.
